### PR TITLE
Capstone project - Javascript

### PR DESCRIPTION
**Pull request by serengia**
Hello.
Kindly review the request to merge development branch to main branch. 
The branch contains the following features:
- Web app UI implementation.
- Functionality to add and fetch menus from an API
- Functionality to add and fetch comments from an API
- Testing implementation using Jest.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import navigationHandler from "./modules/navigationHandler.js";
+
+const popupHook = document.querySelector(".popup-hook");
+const itemContainer = document.querySelector(".cards");
+const itemCount = document.querySelector(".item-count");
+
+(async () => {
+  const meals = await getAllMeal();
+  const likes = await getLikes();
+  itemCount.textContent = meals.length;
```
Comment (gemmen29) [inline src/index.js:20]: - Kindly make sure to follow the guidelines in the card `Add all items counter on the Homepage - Student A` in this [issues](https://github.com/serengia/capstone-project--consuming-API/issues/6): 
   - [ ] You need to implement the counting logic in a separate module.
   - [ ] A counter function should look for specific DOM elements. 
   - [ ] A counter function should cover all the edge cases you can think about.

File: src/modules/generatePopupMarkup.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                           <p><a target="__blank" href="${
+                             detailedCardData.strYoutube
+                           }">Watch on Youtube</a></p>
+                           <p>Category: ${detailedCardData.strCategory}</p>
+                           <p>Area: ${detailedCardData.strArea}</p>
+                           <p>Meal Id: ${detailedCardData.idMeal}</p>
+                        </div>
+                        <div class="comments-container">
+                            <h3>Comments(<span class="comments-count">${
+                              commentsData.length || 0
```
Comment (gemmen29) [inline src/modules/generatePopupMarkup.js:38]: - Kindly make sure to follow the guidelines in the card `Add comments counter - Student B` in this [issues](https://github.com/serengia/capstone-project--consuming-API/issues/5):
   - [ ] You need to implement the counting logic in a separate module. 
   - [ ] A counter function should look for specific DOM elements. 
   - [ ] A counter function should cover all the edge cases you can think about.

File: main.test.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+const { expect, describe } = require("@jest/globals");
```
Comment (gemmen29) [inline main.test.js:1]: - [ ] Kindly update the test for the count of all items based on the new implementation and please make sure that your test includes all edge cases like if there are 0 items on the page.

Comment (gemmen29) [inline main.test.js:1]: - [ ] Kindly update the test for the count of all comments based on the new implementation and please make sure that your test includes all edge cases like if there are 0 items on the page.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (serengia) [reply src/index.js:20]: Hello @gemmen29 . Thank you for your feedback. I have been struggling for many hours without success to make the changes you requested work. Also, after several attempts, I still do not clearly understand the process I need to follow to implement the changes. 
### My concern
Besides the above-highlighted issues I am facing, I feel the requested changes will go against the DRY and KISS principles. Why do I think so? In my case, I am already passing the comments array (from which I can get the length) in my markup. Having a separate function to calculate and insert the comments count for every item will be unnecessary repetition besides making it harder to read my code.
Kindly advice. I feel stuck.

File: src/modules/generatePopupMarkup.js (hunk 2 above)
Comment (serengia) [reply src/modules/generatePopupMarkup.js:38]: Hello @gemmen29 . Thank you for your feedback. I have been struggling for many hours without success to make the changes you requested work. Also, after several attempts, I still do not clearly understand the process I need to follow to implement the changes. 
### My concern
Besides the above-highlighted issues I am facing, I feel the requested changes will go against the DRY and KISS principles. Why do I think so? In my case, I am already passing the comments array (from which I can get the length) in my markup. Having a separate function to calculate and insert the comments count for every item will be unnecessary repetition besides making it harder to read my code.
Kindly advice. I feel stuck.

Comment (gemmen29) [reply src/modules/generatePopupMarkup.js]: Hi @serengia You need to count items from your page DOM instead of getting it directly from the API. You can do this by counting the DOM elements that have something similar like for example: `Class name`

### Hint: you need to use the Document object for more information you can see the documentation from [here](https://developer.mozilla.org/en-US/docs/Web/API/Document).

Comment (serengia) [reply src/modules/generatePopupMarkup.js]: Thank you @gemmen29  for getting back. We've been able to implement the change.

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @serengia @gemmen29    👋 
 
 # STATUS: APPROVED ✔️  🍾  🥳

 Your project is complete! There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉

 
 
 ## Highlights
✔️  Proper use of GitHub features 💯 
✔️ Linters are passing 👏 
✔️ Godd design 💯 
 
 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. The previous reviewer missed some of them, and addressing them will improve your application._
 
 
 Cheers and Happy coding!👏👏👏
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
